### PR TITLE
[IOTDB-968] Support time predicate in select last

### PR DESCRIPTION
**Pull request by wshao08**
JIRA issue: https://issues.apache.org/jira/browse/IOTDB-968

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java [hunk 1]
```diff
@@ -127,103 +127,64 @@ protected TimeValuePair calculateLastPairForOneSeries(
...
   }
 
-  private static TimeValuePair constructLastPair(long timestamp, Object value,
-                                                 TSDataType dataType) {
-    return new TimeValuePair(timestamp, TsPrimitiveType.getByType(dataType, value));
+  private static boolean satisfyFilter(Filter filter, TimeValuePair tvPair) {
+    if (filter != null &&
+        filter.satisfy(tvPair.getTimestamp(), tvPair.getValue().getValue())) {
+      return true;
+    }
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java:187]: if the filter is null, shoudn't we return true?

File: server/src/main/java/org/apache/iotdb/db/query/executor/fill/LastPointReader.java [hunk 2]
```diff
@@ -216,6 +216,9 @@ private boolean shouldUpdate(long time, long version, long newTime, long newVers
   }
 
   private boolean endtimeContainedByTimeFilter(Statistics statistics) {
+    if (timeFilter == null) {
+      return false;
+    }
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/executor/fill/LastPointReader.java:221]: Same as before, if the filter is null, shoudn't we return true?

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java (hunk 1 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java]: Fixed

File: server/src/main/java/org/apache/iotdb/db/query/executor/fill/LastPointReader.java (hunk 2 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/query/executor/fill/LastPointReader.java]: Fixed

**Review by JackieTien97 — APPROVED** [review]

**Review by HTHou — APPROVED** [review]

**Pull request merged**